projects/onto-search/src/lib/onto-search-facet/range-facet/range-histogram/range-histogram.component.ts

```diff
@@ -0,0 +1,116 @@
+import {AfterViewInit, Component, ElementRef, Input, ViewChild} from '@angular/core';
+import {SearchFacetGroupModel} from '../../models/search-facet-group-model';
+import {SearchFacetModel} from '../../models/search-facet-model';
+import {Observable} from 'rxjs';
+import {HistogramModel} from '../models/histogram-model';
+
+@Component({
+  selector: 'onto-range-histogram',
+  templateUrl: './range-histogram.component.html',
+  styleUrls: ['./range-histogram.component.scss']
+})
+export class RangeHistogramComponent implements AfterViewInit {
+  @Input()
+  public data: SearchFacetGroupModel;
+
+  @Input()
+  public containerWidth: Observable<number>;
+
+  @Input()
+  private histogramModel?: HistogramModel;
+
+  @ViewChild('canvas')
+  private canvas: ElementRef;
+
+  private ctx;
+  private width;
+
+  public ngAfterViewInit(): void {
+    this.containerWidth.subscribe((width) => {
+      this.width = width;
+      this.init();
+    });
+  }
+
+  init():void {
+    this.fitToContainer(this.canvas.nativeElement);
+    this.ctx = this.canvas.nativeElement.getContext('2d');
+
+    this.data.facetGroupData.forEach((e) => e);
+    const data: any[] = JSON.parse(JSON.stringify(this.data.facetGroupData));
```

Comment:
This is strange. Please clarifty! Also these JSON methods potentially can throw exceptions

Reply:
fixed